src/snowflake/cli/_plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py

```diff
@@ -48,28 +51,77 @@ def _convert_v2_artifact_to_v1_dict(
     return v2_artifact
 
 
-def _pdf_v2_to_v1(v2_definition: DefinitionV20) -> DefinitionV11:
+def _pdf_v2_to_v1(
+    v2_definition: DefinitionV20,
+    package_entity_id: str = "",
+    app_entity_id: str = "",
+) -> DefinitionV11:
     pdfv1: Dict[str, Any] = {"definition_version": "1.1", "native_app": {}}
 
     app_package_definition: Optional[ApplicationPackageEntityModel] = None
     app_definition: Optional[ApplicationEntityModel] = None
 
+    # Enumerate all application package and application entities in the project definition
+    packages = {}
+    apps = {}
     for key, entity in v2_definition.entities.items():
-        if entity.get_type() == ApplicationPackageEntityModel.get_type():
-            if app_package_definition:
-                raise ClickException(
-                    "More than one application package entity exists in the project definition file."
-                )
-            app_package_definition = entity
-        elif entity.get_type() == ApplicationEntityModel.get_type():
-            if app_definition:
+        entity_type = entity.get_type()
+        if entity_type == ApplicationPackageEntityModel.get_type():
+            packages[key] = entity
+        elif entity_type == ApplicationEntityModel.get_type():
+            apps[key] = entity
+
+    # Determine the application entity to convert, there can be zero or one
+    if app_entity_id:
+        # If the user specified an app entity ID, use that one directly
+        app_definition = apps.get(app_entity_id)
+    elif len(apps) == 1:
+        # Otherwise, if there is only one app entity, fall back to that one
+        app_definition = next(iter(apps.values()))
+    elif len(apps) > 1:
+        # If there are multiple app entities, the user must specify which one to use
+        raise ClickException(
+            "More than one application entity exists in the project definition file, "
+            "specify --app-entity-id to choose which one to operate on."
+        )
```

Comment:
I think it's worth mentioning in comments that while we could support `snow app [bundle|deploy] --package-entity-id=pkg`, we're choosing not to in order to simplify the implementation.

Reply:
Aren't all commands decorated with this? Why won't bundle/deploy work out of the box?

Reply:
yeah it is, @sfc-gh-cgorrie I'm not sure I understand your comment

![image](https://github.com/user-attachments/assets/c31d7312-5e30-40c4-9b31-2a10791e88cc)

Reply:
My point is that you need _app_ entity specified if you only provide one, even though the bundle and diff commands operate on the package only.

Reply:
hmm it shouldn't be necessary, the app entity remains optional as it was before

Reply:
ah i see what you mean, if there are multiple app entities you'd need to specify which even though the command doesn't use it. I'm open to following up this PR to make the app entity entirely optional per-command

Reply:
Sounds good, let's address in a follow-up.